torch/_inductor/codegen/cpp.py

```diff
@@ -406,11 +406,9 @@ def _print_Round(self, expr):
     def _print_RoundDecimal(self, expr):
         assert len(expr.args) == 2
         number, ndigits = expr.args
+        cpp_type = INDEX_TYPE if isinstance(number, sympy.Integer) else "double"
```

Comment:
We generally use `number.is_integer`, which allows expressions that evaluate to integers whereas `sympy.Integer` means a constant integer.